File: backend.js
// Save this code in backend.js
const express = require('express');
const http = require('http');
const socketIO = require('socket.io');

const app = express();
const server = http.createServer(app);
const io = socketIO(server);

let sharedContent = '';

app.use(express.static(__dirname + '/public'));

app.get('/', (req, res) => {
  res.sendFile(__dirname + '/index.html');
});

io.on('connection', (socket) => {
  socket.emit('content-update', sharedContent);

  socket.on('content-update', (content) => {
    sharedContent = content;
    io.emit('content-update', sharedContent);
  });
});

const port = 3000;
server.listen(port, () => {
  console.log(`Server running on http://localhost:${port}`);
});

File: backend.py
# Save this code in backend.py
from flask import Flask, render_template
from flask_socketio import SocketIO, emit

app = Flask(__name__)
app.config['SECRET_KEY'] = 'secret'
socketio = SocketIO(app)

shared_content = ""

@app.route('/')
def index():
    return render_template('index.html')

@socketio.on('content-update')
def handle_content_update(content):
    global shared_content
    shared_content = content
    emit('content-update', shared_content, broadcast=True)

if __name__ == '__main__':
    socketio.run(app)
